### PR TITLE
Add version v1.24

### PR DESCRIPTION
**Pull request by matttrach**
Adding v1.24.2 to channels-rke2

Leaving out enable-pprof server cli option per this comment: https://github.com/rancher/rke2/issues/3055#issuecomment-1164734343

### CODE REVIEW

**Review by rancher-max — DISMISSED** [review]
LGTM

**Review by kinarashah — DISMISSED** [review]

**Review by jiaqiluo — DISMISSED** [review]

**Comment by brandond** [comment]
I believe the ask was for k3s 1.24 as well; do you want to cover that here, or open a separate PR?

**Comment by nikolaishields** [comment]
@brandond I was just about to open one. @matttrach if you want to pair on this we can just add it to this PR if you'd like.

**Comment by brandond** [comment]
Also, based on what's going on over in https://github.com/rancher/kontainer-driver-metadata/pull/927 it looks like we may need to add another new attribute to these releases to enable secrets encryption rotation. Or alternatively, that PR will need to be rebased on top of this one and updated to add the correct `featureVersions` to the 1.24 releases.

 cc @jakefhyde @Oats87

**Comment by nikolaishields** [comment]
cc @rancher-max

File: channels.yaml [hunk 1]
```diff
@@ -238,3 +238,8 @@ releases:
     maxChannelServerVersion: v2.6.99
     serverArgs: *serverArgs-v3
     agentArgs: *agentArgs-v2
+  - version: v1.24.2+k3s2
+    minChannelServerVersion: v2.6.4-alpha1
```
Comment (rancher-max) [inline channels.yaml]: minChannelServerVersion should be updated to be `v2.6.7-alpha1`

**Review by rancher-max — CHANGES_REQUESTED** [review]

**Comment by rancher-max** [comment]
I think #927 should take care of the secrets encryption. This is likely to merge before, so that PR should probably wait for this, rebase, and update accordingly IMO.

File: channels.yaml (hunk 1 above)
Comment (nikolaishields) [reply channels.yaml]: This is done.

**Review by rancher-max — DISMISSED** [review]

**Review by jiaqiluo — DISMISSED** [review]

**Review by brandond — DISMISSED** [review]

**Comment by snasovich** [comment]
Do we want to update `appDefaults` to make 1.24.x default for 2.6.7 (https://github.com/rancher/kontainer-driver-metadata/blob/76aa4a03283ce2505376c33b4f10da44a3880f5d/channels-rke2.yaml#L4-L10)?
@kinarashah @jiaqiluo , or are you planning to switch defaults separately after initial validations?

**Comment by snasovich** [comment]
> Do we want to update `appDefaults` to make 1.24.x default for 2.6.7 (
> 
> https://github.com/rancher/kontainer-driver-metadata/blob/76aa4a03283ce2505376c33b4f10da44a3880f5d/channels-rke2.yaml#L4-L10
> 
> )?
> @kinarashah @jiaqiluo , or are you planning to switch defaults separately after initial validations?

Update: we'll bump defaults separately.

**Comment by jakefhyde** [comment]
https://github.com/rancher/kontainer-driver-metadata/pull/927 is now merged

**Comment by rancher-max** [comment]
> #927 is now merged

@matttrach @nikolaishields We should probably rebase this now and do the secrets encryption stuff added there.

**Review by dereknola — APPROVED** [review]

**Review by nikolaishields — APPROVED** [review]

**Review by rancher-max — APPROVED** [review]
Looks like this is just RKE2 now. @nikolaishields You'll need to create a separate PR in that case for k3s including what has been discussed on this PR in regards to secrets encryption.

**Comment by nikolaishields** [comment]
That's the plan @rancher-max, I'll follow up with a specific k3s pr.

**Pull request merged**